src/api/apiRoutes.js

```diff
@@ -0,0 +1,36 @@
+import express from "express"
+import { getGames } from "./helpers.js"
+
+const router = express.Router()
+
+router.get("/get-top-products", (req, res) => {
+  const games = getGames()
+  const gamesSortedByDate = games
+    .sort((a, b) => a.releaseDate - b.releaseDate)
+    .slice(0, 3)
+  return res.send(gamesSortedByDate)
+})
+
+// GAME PRODUCTS LIST
+router.get(`/products`, (req, res) => {
+  let listOfGames = getGames()
+  const searchQuery = req.query.search?.toLowerCase() || ""
+  const categoryQuery = req.query.category
+  const filteredByCategory = listOfGames.filter((game) =>
```

Comment:
Are you sure that category is always provided?
I remember if you use search bar then you'll get `?search=bla-bla-bla` and category here isn't presented, so your request would look like `fetch('api/products?search=bla-bla-bla')` and you receive an undefined value for category. Please concider do an undefined check before doing filter

Reply:
I did what you requested. I'm not arguing just want to share my thoughts on that part of code. We use filteredByCategory array only when we pass certain conditions like:  if (categoryQuery && searchQuery) or if (categoryQuery). I got the point this array will be empty if  we have no any category, but in this case we don't use it at all. Is it like the thing to remove possible bugs and be more efficient?